iroha/src/smartcontracts/isi/mod.rs

```diff
@@ -107,6 +107,15 @@ pub enum MathError {
     OverflowError,
 }
 
+#[derive(Debug, Clone)]
+pub struct HashNotFound(pub Hash);
+
+impl Display for HashNotFound {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "Block with parent hash {} not found", self.0)
```

Comment:
It is not parent hash, it is the hash of this block though

Reply:
> 
> 
> It is not parent hash, it is the hash of this block though

No, the error in question is raised when this search will give 0 results:
```
.position(|block_entry| block_entry.value().header().previous_block_hash == hash)
```
So we search any block with this "parent" (previous) hash.

Reply:
Then I think we should name error `ParentHashNotFound`